File: src/utils/dimensions.js
/**
 * Dimensiuni catalog: valori > 10 sunt centimetri → metri.
 */
export function normalizeDim(value, fallback = 1) {
  const n = parseFloat(value)
  if (!Number.isFinite(n) || n <= 0) return fallback
  if (n > 10) return Math.round((n / 100) * 1000) / 1000
  return n
}

export function normalizeProductDims(product) {
  if (!product) return product
  return {
    ...product,
    width: normalizeDim(product.width, 1),
    depth: normalizeDim(product.depth, 0.8),
    height: product.height != null ? normalizeDim(product.height, 1) : product.height,
  }
}

File: backend/furniture/dimensions.py
"""Normalizare dimensiuni catalog: cm → metri pentru plan cameră."""


def normalize_dimension_meters(value, default: float = 1.0) -> float:
    """
    Valori > 10 sunt tratate ca centimetri (ex. 180 → 1.8 m).
    Valori ≤ 10 sunt deja în metri (ex. 1.05, 2.0).
    """
    try:
        v = float(value)
    except (TypeError, ValueError):
        return default
    if v <= 0:
        return default
    if v > 10:
        return round(v / 100.0, 3)
    return round(v, 3)


def normalize_product_dimensions(product: dict) -> dict:
    """Returnează copie cu width/depth/height în metri."""
    out = dict(product)
    out['width'] = normalize_dimension_meters(out.get('width'), 1.0)
    out['depth'] = normalize_dimension_meters(out.get('depth'), 0.8)
    if out.get('height') is not None:
        out['height'] = normalize_dimension_meters(out.get('height'), 1.0)
    return out
